### PR TITLE
Add API for resending CommCare install info over SMS

### PR DESCRIPTION
**Pull request by gcapalbo**
http://manage.dimagi.com/default.asp?237488
best reviewed commit by commit
@benrudolph

### CODE REVIEW

File: corehq/apps/sms/messages.py [hunk 1]
```diff
@@ -87,12 +88,12 @@
         "'join {0} worker [username]', entering your requested username in place of [username]"),
     MSG_REGISTRATION_WELCOME_CASE: ugettext_noop("Thank you for registering with CommCareHQ."),
     MSG_REGISTRATION_WELCOME_MOBILE_WORKER: ugettext_noop("Thank you for registering with CommCareHQ."),
+    MSG_REGISTRATION_INSTALL_COMMCARE: ugettext_noop("To install CommCare, follow this link to the Google Play store: {0}"),
```
Comment (dimagimon) [inline corehq/apps/sms/messages.py]: E501 line too long (124 > 115 characters)

File: corehq/apps/sms/views.py [hunk 2]
```diff
@@ -2119,31 +2120,49 @@ class InvitationAppInfoView(View, DomainViewMixin):
...
+    @property
+    @memoized
+    def odk_url(self):
+        try:
+            odk_url = SelfRegistrationInvitation.get_app_odk_url(self.domain, self.app_id)
+        except Http404:
+            odk_url = None
+
+        if odk_url:
+           return odk_url
```
Comment (dimagimon) [inline corehq/apps/sms/views.py]: E111 indentation is not a multiple of four

File: corehq/apps/sms/resources/v0_5.py [hunk 3]
```diff
@@ -137,6 +154,27 @@ def is_valid(self, bundle, request=None):
         return {}
 
 
+class UserSelfRegistrationReinstallValidation(BaseUserSelfRegistrationValidation):
+
+    def is_valid(self, bundle, request=None):
```
Comment (benrudolph) [inline corehq/apps/sms/resources/v0_5.py]: nit: kind of a weird function name for something that doesn't return a boolean

File: corehq/apps/sms/resources/v0_5.py [hunk 4]
```diff
@@ -227,3 +268,59 @@ def post_list(self, request, **kwargs):
...
+
+    class Meta:
+        authentication = RequirePermissionAuthentication(Permissions.edit_data)
+        resource_name = 'sms_user_registration_reinstall'
+        allowed_methods = ['post']
+        validation = UserSelfRegistrationReinstallValidation()
+
+    def full_hydrate(self, bundle):
+        if not self.is_valid(bundle):
+            raise ImmediateHttpResponse(response=self.error_response(bundle.request, bundle.errors))
```
Comment (benrudolph) [inline corehq/apps/sms/resources/v0_5.py]: should this be returning the errors from is_valid

**Review by benrudolph — APPROVED** [review]
👍

**Comment by benrudolph** [comment]
nothing blocking so gonna go ahead and merge

File: corehq/apps/sms/resources/v0_5.py (hunk 3 above)
Comment (benrudolph) [inline corehq/apps/sms/resources/v0_5.py]: ah gotcha, thanks

File: corehq/apps/sms/resources/v0_5.py (hunk 4 above)
Comment (gcapalbo) [inline corehq/apps/sms/resources/v0_5.py]: yeah that happens automatically - the `is_valid` method being called here is the method on the `Resource` superclass which ends up calling `is_valid` on the custom `Validation` class and puts the errors in `bundle.errors`

**Pull request merged**